### PR TITLE
Feat/add basic nse impl

### PR DESCRIPTION
**Pull request by danielvladco**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 go:
-	GOOS=linux CGO_ENABLED=0 go build -o nse ./cmd/nsed/main.go
+	GO111MODULE=on CGO_ENABLED=0 GOOS=linux go build -ldflags '-extldflags "-static"'-o vl3_nse ./cmd/nsed/main.go
 
 docker:
 	docker build -t cisco/cnns-nse:latest .
 
 kind-load:
-	kind load docker-image cisco/cnns-nse:latest
+	kind load docker-image cisco/cnns-nse:latest --name $(KIND_CsLUSTER)
```
Comment (adodon2go) [inline Makefile]: typo

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
 helm:
 	helm install --name cnns-nse ./k8s/cnns-nse
```
Comment (adodon2go) [inline Makefile]: Add EOL at the end of file

File: cmd/nsed/main.go [hunk 3]
```diff
@@ -1,54 +1,113 @@
...
+	"vl3nse/internal/vppagent"
 	"vl3nse/pkg/nseconfig"
 )
 
 var (
-	configFile = flag.String("cfg", "/etc/cnns-nse/config.yaml", "the path to config file, default: /etc/cnns-nse/config.yaml")
+	configFile = flag.String("config", "/etc/cnns-nse/config.yaml", "the path to config file, default: /etc/cnns-nse/config.yaml")
 
-	logger = log.New(os.Stderr, "test", 0)
+	nsPodIp          = envOrDefault("NSE_POD_IP", "2.2.20.0")
```
Comment (adodon2go) [inline cmd/nsed/main.go]: Note: 
We already have envOrDefault implementation in github.com/dtornow/cnns-nsr/nsrenv package. The repo is private and I can understand why it is not easy to import it directly...

File: pkg/nseconfig/validate.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	for _, err := range v {
+		fmt.Fprintf(b, "\t%s\n", err)
+	}
+	return b.String()
+}
+
+func (c CNNS) validate() error {
+	var errs InvalidConfigErrors
+	if empty(c.Address) {
+		errs = append(errs, fmt.Errorf("cnns addreses is not set"))
```
Comment (adodon2go) [inline pkg/nseconfig/validate.go:22]: addresses

File: scripts/test.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
+echo "$0"
+echo "$1"
+echo "$2"
+echo "$3"
+echo "$@"
+echo "$@"
```
Comment (adodon2go) [inline scripts/test.sh]: why we print all arguments twice?

File: scripts/test.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+echo "$#"
+
+for arg in $@; do
+    case ${arg} in
+        -h|--help)
+            usage
+            exit
```
Comment (adodon2go) [inline scripts/test.sh]: exit 0

File: scripts/test.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        ;;
+        --namespace=?*)
+            echo "nmsp: ${arg#*=}"
+        ;;
+        *)
+            echo "arg: ${arg}"
+        ;;
+    esac
+done
```
Comment (adodon2go) [inline scripts/test.sh]: No EOL here

File: scripts/vl3_interdomain.sh [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+HUB=${HUB:-tiswanso}
+TAG=${TAG:-kind_ci}
+PULLPOLICY=${PULLPOLICY:-IfNotPresent}
+INSTALL_OP=${INSTALL_OP:-apply}
+
+for i in "$@"; do
+    case $i in
+        -h|--help)
+            usage
+            exit
```
Comment (adodon2go) [inline scripts/vl3_interdomain.sh]: exit 0

**Review by adodon2go — APPROVED** [review]

**Review by tiswanso — COMMENTED** [review]
please recreate the PR against the kind_ci branch... we can merge it in and then discuss/resolve the NSE repo branch stuff as a separate action.

**Pull request merged**